Updated Test Components for Perf Tests

labkey-danield opened the pull request:
#### Rationale
I needed to make some minor changes to accommodate the UI Perf tests.

#### Related Pull Requests
* https://github.com/LabKey/sampleManagement/pull/1552

#### Changes
* Added a getter for the grid Pager dropdown menu. The perf test uses it to get a reference to the menu item it needs to click.
* Use a more generic class name in the finder for the success button on the EntityBulkUpdateDialog. It wasn't finding the button when doing a bulk update from the 'Add New' page.
* Added DataClassAPIHelper.getRowIdsForDataEntity. Used to get the row id of an entity in a data class, specifically the row id of a source.

labkey-danield committed:
Add getters for average and calls to CPUTimer.
Added DataClassAPIHelper.getRowIdsForDataEntity to get row ids for sources.
Added Pager.getDropDownMenu to get the dropdown menu so the menu item click can be isolated inside the timers.
Created a base class for the UI perf test (renamed an existing UI perf test file)
Broke out the samples perf test to their own class.
Added source perf tests.

labkey-danield committed:
Merge branch 'develop' into fb_perfTestAutomationPOC

labkey-danield committed:
Merge branch 'develop' into fb_perfTestAutomationPOC

labkey-danield committed:
Added test for creating aliquots and derived samples.
Updated some data to be the same as in the db dump data.
Cleaned up and consolidated some code.
Added a better check after test run.

labkey-danield committed:
Merge branch 'develop' into fb_perfTestAutomationPOC

labkey-tchad commented on src/org/labkey/test/util/exp/DataClassAPIHelper.java:
```diff
@@ -51,4 +60,31 @@ public static List<FieldDefinition> dataClassTestFields()
                 new FieldDefinition("attachmentColumn", FieldDefinition.ColumnType.Attachment));
     }
 
+    public static Map<String, Integer> getRowIdsForDataEntity(String containerPath, String dataTypeName, List<String> entityNames) throws IOException, CommandException
```
Try to use consistent terminology. "Data Class" or "Source" are the widely used terms for things in `exp.data` (we might want to do some standardization across tests, similar to the change from "Sample Set" to "Sample Type").
I'd also name this method to better match the equivalent method for sample types (`SampleTypeAPIHelper.getRowIdsForSamples`). Maybe `getRowIdsForSources`? I think "entity" is a term that can refer to samples and sources.

labkey-danield replied:
Updated to explicitly refer to sources and source types and changed the name to your suggestion.

labkey-jeckels commented on src/org/labkey/test/util/exp/DataClassAPIHelper.java:
```diff
@@ -51,4 +60,31 @@ public static List<FieldDefinition> dataClassTestFields()
+
+        // Check that the names returned from the query match the names sent in.
+        Set<String> names = new HashSet<>(entityNames);
+        Assert.assertTrue("The names of the data entities returned from the query do not match the names sent in.",
```
Should they be an exact match? Or is one a superset of the other? If they're the same, doing an Assert.assertEquals() has the nice benefit of giving more detail about their contents when the check fails.

labkey-danield replied:
Good point, they should be  identical. Changed the check to assertEquals.

labkey-danield committed:
Merge branch 'develop' into fb_perfTestAutomationPOC

labkey-danield committed:
Update labkey schemas to latest version.

labkey-danield committed:
Rename DataclassAPIHelper.getRowIdsForDataEntity to getRowIdsForSources.
Clarified terms/text to indicate method is for sources.
Changed check to assertEquals.

labkey-danield committed:
Changed check to assertEquals in SampleTypeAPIHelper.getRowIdsForSamples

labkey-danield committed:
Merge branch 'develop' into fb_perfTestAutomationPOC
